paasta_tools/setup_kubernetes_job.py

```diff
@@ -103,43 +103,78 @@ def validate_job_name(service_instance: str) -> bool:
     return True
 
 
+def _bounce_service_instance(
+    deploy_strategy: str,
+    app: DeploymentWrapper,
+    kube_client: KubeClient,
+    existing_apps: dict,
+    existing_kube_deployments: set,
+) -> None:
+
+    if deploy_strategy == "brutal":
+        app.deep_delete(kube_client)
+        app.create(kube_client)
+        return
+    if (
+        app
+        and (app.kube_deployment.service, app.kube_deployment.instance)
+        not in existing_apps
+    ):
+        app.create(kube_client)
+    elif app and app.kube_deployment not in existing_kube_deployments:
+        app.update(kube_client)
+    else:
+        log.debug(f"{app} is up to date, no action taken")
+
+
 def setup_kube_deployments(
     kube_client: KubeClient,
     service_instances: Sequence[str],
     soa_dir: str = DEFAULT_SOA_DIR,
 ) -> bool:
+
     if service_instances:
         existing_kube_deployments = set(list_all_deployments(kube_client))
         existing_apps = {
             (deployment.service, deployment.instance)
             for deployment in existing_kube_deployments
         }
+
     service_instances_with_valid_names = [
         decompose_job_id(service_instance)
         for service_instance in service_instances
         if validate_job_name(service_instance)
     ]
-    applications = [
-        create_application_object(
-            kube_client=kube_client,
-            service=service_instance[0],
-            instance=service_instance[1],
-            soa_dir=soa_dir,
+
+    job_configs_apps = {}
+
+    for service_instance in service_instances_with_valid_names:
+        service_name, instance_name, _, _ = service_instance
+        cluster = load_system_paasta_config().get_cluster()
+        job_kubernetes_config = load_kubernetes_service_config_no_cache(
+            service_name, instance_name, cluster, soa_dir=soa_dir
+        )
+        job_configs_apps[service_instance] = (
+            job_kubernetes_config,
+            create_application_object(
+                kube_client=kube_client,
+                service=service_name,
+                instance=instance_name,
+                soa_dir=soa_dir,
+            ),
         )
-        for service_instance in service_instances_with_valid_names
-    ]
 
-    for _, app in applications:
-        if (
-            app
-            and (app.kube_deployment.service, app.kube_deployment.instance)
-            not in existing_apps
-        ):
-            app.create(kube_client)
-        elif app and app.kube_deployment not in existing_kube_deployments:
-            app.update(kube_client)
-        else:
-            log.debug(f"{app} is up to date, no action taken")
```

Comment:
Instead of changing this file, I recommend you add 
```
    if self.soa_configs. get_bounce_method() == "brutal":
        app.deep_delete(kube_client)
```
in DeploymentWrapper.  You don't need to change paasta_tools/setup_kubernetes_job.py  at all.

Reply:
I forgot to mention. I'm not sure if deep_delete actually wait for the deletion to finish or not. And I'm not sure if that's what you expect. If you want it to wait for deletion to finish, I suggest you to create a new kubeclient and use a seperate thread for this. So it does not block other actions.

Reply:
Deep_delete doesnt wait and if we dont wait , we cannot call create, Do you recommend a background (delete+create) so the client doenst have to wait... or (delete+ wait for delete to finish+create)
Also not exactly sure where to add that if statement in the DeploymentWrapper class?

Reply:
@felfilali I recommend you add another delete function that waits, and also run delete + create async so it doesn't block. So (delete+ wait for delete to finish+create) async.

You can just modify create method in DeploymentWrapper class. line 192.